File: frontend/src/api/agents.ts
/**
 * 创建时间: 2026-06-11
 * 作者: hongchuwudi
 * 描述: Agent 管理 API — 实时状态查询 + 提示词增删改查 + 决策历史
 *
 * 包含:
 * - fetchAgentStatus     — 获取所有 Agent 的当前运行状态
 * - fetchPrompts         — 获取所有 Agent 的提示词列表
 * - updatePrompt         — 更新单个 Agent 的提示词
 * - resetPrompt          — 重置单个 Agent 提示词为默认值
 * - reloadAgents         — 提示词变更后重新加载所有 Agent 实例
 * - fetchDecisions       — 分页查询 Agent 决策历史记录
 */

import { get, put, post } from '../utils/request'
import type { AgentStatusResponse, AgentPrompt, AgentDecisionPage } from '../types/agents'

// ── Agent 状态 ─────────────────────────────────────────

// 获取所有 Agent 的当前运行状态：调度师/分析师/复盘师/风控师/交易裁决员
// 返回各 Agent 的最新输入、输出、移交目标、时间戳
export const fetchAgentStatus = () => get<AgentStatusResponse>('/agents/status')

// ── 提示词管理 ────────────────────────────────────────

// 获取所有 Agent 的提示词列表，包含名称、版本号、内容文本
export const fetchPrompts = () => get<AgentPrompt[]>('/agents/prompts')

// 更新单个 Agent 的提示词文本，name 为提示词文件名（不含扩展名）
// 保存后版本号自动递增，需调用 reloadAgents 使变更生效
export const updatePrompt = (name: string, text: string) =>
  put<{ ok: boolean }>(`/agents/prompts/${name}`, { text })

// 重置单个 Agent 的提示词为系统默认值（从代码模板恢复）
export const resetPrompt = (name: string) =>
  post<{ ok: boolean }>(`/agents/prompts/${name}/reset`)

// 提示词变更后重新加载所有 Agent 实例，使新提示词立即生效
// 注意：运行中的 tick 不受影响，下一个 tick 开始使用新提示词
export const reloadAgents = () =>
  post<{ ok: boolean; message: string }>('/agents/reload')

// ── 决策历史 ──────────────────────────────────────────

// 分页查询 Agent 决策历史记录，每 tick 存一条
// 返回 { data, page, page_size, total, total_pages }
// 每条记录含 signal/confidence/reason/stop_loss/take_profit/mode
export const fetchDecisions = (page = 1, pageSize = 20) =>
  get<AgentDecisionPage>(`/agents/decisions?page=${page}&page_size=${pageSize}`)

File: backend/app/api/v1/agents.py
"""
创建时间: 2026-06-07
作者: hongchuwudi
文件名: agents.py Agent状态API
描述: Agent 状态查询 + 提示词热加载管理

包含:
- 端点: GET /api/agents/status — 返回 Agent 最新状态
- 端点: GET /api/agents/prompts — 获取全部提示词
- 端点: PUT /api/agents/prompts/{name} — 更新单个提示词
- 端点: POST /api/agents/prompts/{name}/reset — 重置为默认值
"""

from fastapi import APIRouter, Body, HTTPException

from app.services.agent.agent_status_service import agent_status_service
from app.schemas.responses.agent import AgentStatusResponse

router = APIRouter(prefix="/api/v1/agents", tags=["agents"])


# 返回每个 Agent 的最新状态（输入/输出/移交/工具调用）+ 最近 50 条事件历史。
@router.get("/status", response_model=AgentStatusResponse)
def api_get_agents_status():
    return agent_status_service.get_status()


# === 提示词管理 ===============================================================

@router.get("/prompts")
async def api_get_prompts():
    from app.agents.prompts import get_all_prompts
    return await get_all_prompts()


@router.put("/prompts/{name}")
async def api_update_prompt(name: str, body: dict = Body(...)):
    from app.agents.prompts import set_prompt
    ok = await set_prompt(name, body.get("text", ""))
    if not ok:
        raise HTTPException(400, f"未知的 Agent: {name}")
    return {"ok": True, "message": f"{name} 提示词已更新"}


@router.post("/prompts/{name}/reset")
async def api_reset_prompt(name: str):
    from app.agents.prompts import reset_prompt
    ok = await reset_prompt(name)
    if not ok:
        raise HTTPException(400, f"未知的 Agent: {name}")
    return {"ok": True, "message": f"{name} 提示词已重置"}


@router.post("/reload")
async def api_reload_agents():
    from app.services.engine.engine_control import force_reload_agents
    ok = await force_reload_agents()
    return {"ok": ok, "message": "Agent 已热重载" if ok else "引擎未运行，变更将在下一 tick 生效"}


@router.get("/decisions")
def api_get_decisions(page: int = 1, page_size: int = 20):
    from app.agents.status import query_agent_decisions
    return query_agent_decisions(page, page_size)
